pkg/kubectl/cmd/update.go

```diff
@@ -42,25 +46,37 @@ Examples:
   $ kubectl update pods my-pod --patch='{ "apiVersion": "v1beta1", "desiredState": { "manifest": [{ "cpu": 100 }]}}'
   <update a pod by downloading it, applying the patch, then updating, requires apiVersion be specified>`,
 		Run: func(cmd *cobra.Command, args []string) {
-			filename := GetFlagString(cmd, "filename")
+			mapper, typer := f.Object(cmd)
+			r := resource.NewBuilder(mapper, typer, ClientMapperForCommand(cmd, f)).
+				ContinueOnError().
+				FilenameParam(flags.Filenames...).
+				Flatten().
+				Do()
+
 			patch := GetFlagString(cmd, "patch")
-			if len(filename) == 0 && len(patch) == 0 {
+			if len(flags.Filenames) == 0 && len(patch) == 0 {
 				usageError(cmd, "Must specify --filename or --patch to update")
 			}
-			if len(filename) != 0 && len(patch) != 0 {
+			if len(flags.Filenames) != 0 && len(patch) != 0 {
 				usageError(cmd, "Can not specify both --filename and --patch")
 			}
-			var name string
-			if len(filename) > 0 {
-				name = updateWithFile(cmd, f, filename)
+			if len(flags.Filenames) > 0 {
+				err := r.Visit(func(info *resource.Info) error {
+					if err := updateWithFiles(cmd, f, info); err != nil {
+						return err
+					}
+					fmt.Fprintf(out, "%s\n", info.Name)
+					return nil
+				})
+				checkErr(err)
 			} else {
-				name = updateWithPatch(cmd, args, f, patch)
```

Comment:
Patch is an operation on arbitrary json.  Where that json comes from is irrelevant.


> On Feb 2, 2015, at 3:28 AM, Martin Nagy <notifications@github.com> wrote:
> 
> In pkg/kubectl/cmd/update.go:
> 
> >  			} else {
> > -				name = updateWithPatch(cmd, args, f, patch)
> I can, no problem. But before I do, just to make sure: You do know that this would be a totally new feature, in the sense that patch mode now works by patching downloaded objects and not files provided by command line, right?
> 
> —
> Reply to this email directly or view it on GitHub.
>